python/kfserving/kfserving/api/kf_serving_client.py

```diff
@@ -167,6 +169,68 @@ def patch(self, name, kfservice, namespace=None, watch=False, timeout_seconds=60
         else:
             return outputs
 
+    def replace(self, name, kfservice, namespace=None, watch=False, timeout_seconds=600): # pylint:disable=too-many-arguments,inconsistent-return-statements
+        """Replace the created KFService in the specified namespace"""
+
+        if namespace is None:
+            namespace = utils.set_kfsvc_namespace(kfservice)
+
+        current_kfsvc = self.get(name, namespace=namespace)
+        current_resource_version = current_kfsvc['metadata']['resourceVersion']
+        if kfservice.metadata.resource_version is None:
+            kfservice.metadata.resource_version = current_resource_version
+
+        try:
+            outputs = self.api_instance.replace_namespaced_custom_object(
+                constants.KFSERVING_GROUP,
+                constants.KFSERVING_VERSION,
+                namespace,
+                constants.KFSERVING_PLURAL,
+                name,
+                kfservice)
+        except client.rest.ApiException as e:
+            raise RuntimeError(
+                "Exception when calling CustomObjectsApi->replace_namespaced_custom_object:\
+                 %s\n" % e)
+
+        if watch:
+            kfsvc_watch(
+                name=outputs['metadata']['name'],
+                namespace=namespace,
+                timeout_seconds=timeout_seconds)
+        else:
+            return outputs
+
+
+    def promote(self, name, namespace=None, watch=False, timeout_seconds=600): # pylint:disable=too-many-arguments,inconsistent-return-statements
+        """Promote the created KFService in the specified namespace"""
+
+        if namespace is None:
+            namespace = utils.get_default_target_namespace()
+
+        current_kfsvc = self.get(name, namespace=namespace)
+        api_version = current_kfsvc['apiVersion']
+
+        try:
+            current_canary_spec = current_kfsvc['spec']['canary']
```

Comment:
what is the box character in 'spec'?

Reply:
that's a key for kfserivce from `get` API.